src/data.js

```diff
@@ -5,40 +5,55 @@ const sanitize = require('sanitize-filename');
 const unusedFilename = require('unused-filename');
 const { DATADIR } = require('./dirs');
 const { authMiddleware } = require('./authentication');
+const fs = require('fs');
+const { timeStamp, completionCode } = require('./serverUtils');
+
+var DataRecord = require('./models/DataRecord.model');
+
+const collection = 'experiment_data'
 
 module.exports = (router = new Router()) => {
   router.get('/', authMiddleware);
   router.get('/', (req, res) => {
-    res.status(200).set({
-      'Content-Type': 'application/zip',
-      'Content-disposition': 'attachment; filename=data.zip',
+    DataRecord.find({}).then(function(dataRecords) {
+      var response = {};
+      var count = 1;
+      dataRecords.forEach(function(record) { 
+        response[count] = JSON.parse(record["data"]);
+        count += 1;
+      });
+      return Promise.all([response]);
+    }).then(function(respArray) {
+      res.status(200).set({
+      'Content-Type': 'application/json',
+      });
+      res.send(respArray[0]);
+    }).catch(function(error) {
+      res.status(500).send(error);
     });
-    const zip = archiver('zip');
-    zip.pipe(res);
-    zip.directory(DATADIR, false);
-    zip.finalize();
   });
 
   router.post('/', (req, res) => {
-    const { files: { file } = {} } = req;
-    if (!file) {
-      return res.status(400).send('Collected data file was not uploaded!');
+    const jsonData = req.body
+    if (Object.keys(jsonData).length === 0) {
+      return res.status(400).send('Malformed or bad data');
     }
-    if (!file.name) {
-      return res.status(400).send('Collected data file missing name');
-    }
-    // Sanitize filename to ensure it's a valid path (and avoid certain hacks).
-    const safeName = sanitize(file.name);
-    unusedFilename(path.join(DATADIR, safeName))
-      .then((filename) => {
-        // Store the file on the server.
-        file.mv(filename, (err) => {
-          if (err) {
-            return res.status(500).send(err);
-          }
-          return res.send('Data saved');
-        });
-      });
+    //we need the front end to tell us which config it used to run the experiment, since 
```

Comment:
Can we read in the configID through a route parameter instead of putting it into the JSON? I think that makes more sense.
https://expressjs.com/en/guide/routing.html Check the "route parameters" section

ex: `/data/4` submits data for an experiment running config 4.

Reply:
I feel like the `:id` representing the `config` in `/data/:id`, could kind of be misleading. We can definitely add another route `/data/:id` which would yield the data from the experiment iteration where `experimentID == :id`--and `/data` would just get you the data from all iterations of that particular experiment. But, I think that the `configID` should still be  in the `req.body`--and it would represent a foreign key to the `configs` collection. Let me know what your thoughts on this are. Happy to discuss more